src/rules/selector-no-redundant-nesting-selector/__tests__/index.js

```diff
@@ -137,6 +160,53 @@ testRule(rule, {
     description: "when an amperand precedes a class",
   }, {
     code: `
+      .foo {
+        &bar {
+          & .class {}
+        }
+      }
+    `,
+    line: 4,
+    message: messages.rejected,
+    description: "when an ampersand is used to combine two parts of a classname and an amperand precedes a class",
+  }, {
+    code: `
+      p {
+        & span .class {}
+      }
+    `,
+    line: 3,
+    message: messages.rejected,
+    description: "when an amperand precedes an element and a class",
+  }, {
+    code: `
+      p {
+        & span > .class {}
+      }
+    `,
+    line: 3,
+    message: messages.rejected,
+    description: "when an amperand precedes an element, a child selector and a class",
+  }, {
+    code: `
+      p {
+        & + .foo {}
+      }
+    `,
+    line: 3,
+    message: messages.rejected,
+    description: "when an ampersand precedes a sibling operator",
+  }, {
+    code: `
+      p {
+        & ~ .foo {}
+      }
+    `,
+    line: 3,
+    message: messages.rejected,
+    description: "when an ampersand precedes a sibling operator",
+  }, {
+    code: `
       p {
         &  .class {}
```

Comment:
I pulled a fix for that: #40 If nothing is left out, you can merge it and publish a patch (as per a suggestion in #38